contracts/StakingDistributor.sol

```diff
@@ -1,211 +1,200 @@
 // SPDX-License-Identifier: AGPL-3.0
-pragma solidity ^0.7.5;
-
-import "./libraries/SafeERC20.sol";
-import "./libraries/SafeMath.sol";
+pragma solidity ^0.8.10;
+pragma abicoder v2;
 
 import "./interfaces/IERC20.sol";
 import "./interfaces/ITreasury.sol";
-import "./interfaces/IDistributor.sol";
+import "./interfaces/IStaking.sol";
+import "./interfaces/IUniswapV2Pair.sol";
 
 import "./types/OlympusAccessControlled.sol";
 
-contract Distributor is IDistributor, OlympusAccessControlled {
-    /* ========== DEPENDENCIES ========== */
+/// @notice Updated distributor adds the ability to mint and sync
+///         into Uniswap V2-style liquidity pools, removing the
+///         opportunity-cost dilemma of providing liquidity for
+///         OHM, as well as patches a small bug in the staking contract
+///         that pulls forward an amount of the next epoch rewards. Note that
+///         this implementation bases staking reward distributions on
+///         staked supply, whereas previous implementations used total supply.
+contract Distributor is OlympusAccessControlled {
+
+    error Only_Staking();
+    error Not_Unlocked();
+    error Sanity_Check();
+    error Adjustment_Limit();
+    error Adjustment_Overflow();
+    error Not_Permissioned();
 
-    using SafeMath for uint256;
-    using SafeERC20 for IERC20;
+    struct Adjust {
+        bool add; // whether to add or subtract from the reward rate
+        uint256 rate; // the amount to add or subtract per epoch
+        uint256 target; // the resulting reward rate
+    }
 
     /* ====== VARIABLES ====== */
 
+    /// The OHM Token
     IERC20 private immutable ohm;
+    /// The Olympus Treasury
     ITreasury private immutable treasury;
+    /// The OHM Staking Contract
     address private immutable staking;
 
-    mapping(uint256 => Adjust) public adjustments;
-    uint256 public override bounty;
-
-    uint256 private immutable rateDenominator = 1_000_000;
+    /// The % to increase balances per epoch
+    uint256 public rewardRate;
+    /// Liquidity pools to receive rewards
```

Comment:
Not necessarily relevant to getting this out for M&S, but there may be other cases in the future where a contract is holding naked OHM and could/should? receive rebases. You could generalize this to "receivers", but the interfaces may not conform to the "sync".

Reply:
we might want to implement this this functionality on the heart or tangential contract. i dont think we would need it before then